bazel_integration_test/private/bazel_binaries.bzl

```diff
@@ -99,6 +121,10 @@ def bazel_binaries(versions):
         versions: A `list` of Bazel versions.
     """
     for version in versions:
-        name = "build_bazel_bazel_" + version.replace(".", "_")
-        if not native.existing_rule(name):
+        name = integration_test_utils.bazel_binary_repo_name(version)
+        if native.existing_rule(name):
+            continue
+        if integration_test_utils.is_version_file(version):
```

Comment:
maybe it would be easier to understand if we had two attributes version or version_file?  the distinction would be more apparent.

Reply:
I am trying to support being able to specify `//:.bazelversion` in the `bazel_versions.bzl`. If we allow that syntax in the version string, then we need some code to parse it to decide what the string represents. I opted to put it in the `bazel_binaries` macro so that I could leverage the `version` and `version_file` attributes on `bazel_binary`. This prevents the special case from being propagated deep into the codebase.

So, my questions are:
1. Do we like the being able to specify `//:.bazelversion` in the `bazel_versions.bzl` file?
2. If not, do you have any thoughts on how to allow a client to easily specify the file to use?
3. If we do like the syntax, is there a better way to encapsulate it?